### ISSUE TITLE
Video support?

### ISSUE THREAD

**Issue by kflak** [vydd/sketch#163]
Hi,

I'm dipping my toes into the lisp waters (again) and wondering if there's any way to get video input/playback working with sketch?

**Comment by Kevinpgalligan** [comment]
Not currently, no. I've been looking to add support for saving a video recording of a sketch, but haven't gotten around to it yet. Dunno what it would take to be able to play videos in a sketch -- from what I can tell, the support for video is lacking in the CL ecosystem.

**Comment by Kevinpgalligan** [comment]
These are the currently supported resource types:

```
((:png :jpg :jpeg :tga :gif :bmp) :image)
  ((:ttf :otf) :typeface))
```

A workaround would be to use a gif, or dump the frames of your video as PNG files and cycle through them.

**Comment by kflak** [comment]
Thanks! I am looking at doing live video processing of a webcam, which would be tricky to do with that approach (although there might be some way to get around it with ffmpeg/v4l2...?). For video files the frame dump option could be plausible. I guess that's what processing and openFrameworks does when it comes down to it: displaying one image after another... Although that approach would still need some way to make sure that the video playback is happening at the right speed.

**Comment by Kevinpgalligan** [comment]
Yeah, maybe it could work with ffmpeg/v4l2, though I don't have experience with those and don't know if there would be an efficient way to dump their output into Sketch's OpenGL buffers.
